sdk/services/kwentaToken.ts

```diff
@@ -0,0 +1,428 @@
+			KwentaStakingRewards,
+			KwentaToken,
+			SupplySchedule,
+			vKwentaToken,
+			veKwentaToken,
+			MultipleMerkleDistributor,
+		} = this.sdk.context.mutliCallContracts;
+
+		if (
+			!RewardEscrow ||
+			!KwentaStakingRewards ||
+			!KwentaToken ||
+			!SupplySchedule ||
+			!vKwentaToken ||
+			!MultipleMerkleDistributor ||
+			!veKwentaToken ||
+			!vKwentaRedeemer ||
+			!veKwentaRedeemer
+		) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		const [
+			rewardEscrowBalance,
+			stakedNonEscrowedBalance,
+			stakedEscrowedBalance,
+			claimableBalance,
+			kwentaBalance,
+			weekCounter,
+			totalStakedBalance,
+			vKwentaBalance,
+			vKwentaAllowance,
+			kwentaAllowance,
+			epochPeriod,
+			veKwentaBalance,
+			veKwentaAllowance,
+		]: ethers.BigNumber[] = await this.sdk.context.multicallProvider.all([
+			RewardEscrow.balanceOf(this.sdk.context.walletAddress),
+			KwentaStakingRewards.nonEscrowedBalanceOf(this.sdk.context.walletAddress),
+			KwentaStakingRewards.escrowedBalanceOf(this.sdk.context.walletAddress),
+			KwentaStakingRewards.earned(this.sdk.context.walletAddress),
+			KwentaToken.balanceOf(this.sdk.context.walletAddress),
+			SupplySchedule.weekCounter(),
+			KwentaStakingRewards.totalSupply(),
+			vKwentaToken.balanceOf(this.sdk.context.walletAddress),
+			vKwentaToken.allowance(this.sdk.context.walletAddress, vKwentaRedeemer.address),
+			KwentaToken.allowance(this.sdk.context.walletAddress, KwentaStakingRewards.address),
+			MultipleMerkleDistributor.distributionEpoch(),
+			veKwentaToken.balanceOf(this.sdk.context.walletAddress),
+			veKwentaToken.allowance(this.sdk.context.walletAddress, veKwentaRedeemer.address),
+		]);
+
+		return {
+			rewardEscrowBalance: wei(rewardEscrowBalance),
+			stakedNonEscrowedBalance: wei(stakedNonEscrowedBalance),
+			stakedEscrowedBalance: wei(stakedEscrowedBalance),
+			claimableBalance: wei(claimableBalance),
+			kwentaBalance: wei(kwentaBalance),
+			weekCounter: Number(weekCounter),
+			totalStakedBalance: wei(totalStakedBalance),
+			vKwentaBalance: wei(vKwentaBalance),
+			vKwentaAllowance: wei(vKwentaAllowance),
+			kwentaAllowance: wei(kwentaAllowance),
+			epochPeriod: Number(epochPeriod),
+			veKwentaBalance: wei(veKwentaBalance),
+			veKwentaAllowance: wei(veKwentaAllowance),
+		};
+	}
+
+	public async getEscrowData() {
+		const { RewardEscrow } = this.sdk.context.mutliCallContracts;
+
+		if (!RewardEscrow) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		const schedules = await ((RewardEscrow as unknown) as IRewardEscrow).getVestingSchedules(
+			this.sdk.context.walletAddress,
+			0,
+			DEFAULT_NUMBER_OF_FUTURES_FEE
+		);
+
+		const vestingSchedules = schedules.filter((schedule: any) => schedule.escrowAmount.gt(0));
+
+		const calls = vestingSchedules.map((schedule: any) =>
+			RewardEscrow.getVestingEntryClaimable(this.sdk.context.walletAddress, schedule.entryID)
+		);
+
+		const vestingEntries: {
+			quantity: ethers.BigNumber;
```

Comment:
Would be nice to import BigNumber directly so we don't have to do ethers.BigNumber everywhere

Reply:
Sure. Change pushed.